sdk/core/core-https/src/nodeHttpsClient.ts

```diff
@@ -198,9 +198,11 @@ class NodeHttpsClient implements HttpsClient {
     const proxySettings = request.proxySettings;
     if (proxySettings) {
       if (!this.proxyAgent) {
+        const parsedUrl = new URL(proxySettings.host);
         const proxyAgentOptions: HttpsProxyAgentOptions = {
-          host: proxySettings.host,
+          host: parsedUrl.host,
```

Comment:
Since `port` is passed separately, I think it'd make more sense if we set hostname here instead (and the implementation seems to prefer it: https://github.com/TooTallNate/node-https-proxy-agent/blob/c30a9dc75a272591a80b9b4acef1496f80824f88/src/agent.ts#L53 )

```suggestion
          hostname: parsedUrl.hostname,
```

Reply:
Yes `hostname` is better. `host` worked for my testing so I didn't pay enough attention.